tests/upgrades/test_hostgroup.py

```diff
@@ -16,155 +16,166 @@
+        location=[loc.id], organization=[org.id], name=request.node.name + '_parent_host_grp'
+    ).create()
+    lc_env = entities.LifecycleEnvironment(
+        name=request.node.name + "_lce", organization=org
+    ).create()
+
+    domain = entities.Domain(name=request.node.name + "_domain").create()
+    architecture = entities.Architecture().create()
+    ptable = entities.PartitionTable().create()
+    operatingsystem = entities.OperatingSystem(
+        architecture=[architecture], ptable=[ptable], name=request.node.name + "_os"
+    ).create()
+    medium = entities.Media(operatingsystem=[operatingsystem]).create()
+    subnet = entities.Subnet(
+        location=[loc], organization=[org], name=request.node.name + "_subnet"
+    ).create()
+
+    host_group = entities.HostGroup(
+        architecture=architecture,
+        domain=domain,
+        location=[loc.id],
+        medium=medium,
+        name=request.node.name + "_host_grp",
+        operatingsystem=operatingsystem,
+        organization=[org.id],
+        ptable=ptable,
+        puppet_proxy=proxy,
+        puppet_ca_proxy=proxy,
+        subnet=subnet,
+        parent=parent_hostgroup,
+        lifecycle_environment=lc_env,
+        content_source=proxy,
+        root_pass='rootpass',
+    ).create()
+    assert host_group.name == request.node.name + "_host_grp"
+
+
+@post_upgrade(depend_on=test_pre_create_hostgroup)
+def test_post_crud_hostgroup(request):
+    """Hostgroup is intact post upgrade and update/delete/clone hostgroup
+
+    :id: postupgrade-79958754-94b6-4bfe-af12-7d4031cd2dd2
+
+    :steps:
+
+        1. After upgrade, check hostgroup entities.
+        2. Update existing hostgroup with new entities
+        3. Clone hostgroup.
+        4. Delete hostgroup, parent hostgroup, cloned hostgroup, domain, subnet, os, loc, org.
+
+    :expectedresults: After upgrade
+        1- Hostgroup entities remain same.
+        2- Hostgroup entties update should work.
+        3- Hostgroup cloned should work.
+        4- Cloned hostgroup should be the subset of original hostgroup.
+        5- Hostgroup entities deletion should work
+
+    """
+    pre_test_name = [
+        mark.kwargs['depend_on'].__name__
+        for mark in request.node.own_markers
+        if 'depend_on' in mark.kwargs
+    ][0]
+
+    # verify host-group is intact after upgrade
+    org = entities.Organization().search(query={'search': f'name="{pre_test_name}_org"'})[0]
+    request.addfinalizer(org.delete)
+    loc = entities.Location().search(query={'search': f'name="{pre_test_name}_loc"'})[0]
+    request.addfinalizer(loc.delete)
+    proxy = entities.SmartProxy().search(
+        query={'search': f'url = https://{settings.server.hostname}:9090'}
+    )[0]
+    hostgrp = entities.HostGroup().search(query={'search': f'name={pre_test_name}_host_grp'})[0]
+    request.addfinalizer(hostgrp.parent.delete)
+    request.addfinalizer(hostgrp.delete)
+    assert pre_test_name + "_host_grp" == hostgrp.name
```

Comment:
Again same in post, to set the var for pre-_test_name and use with f-string.

Reply:
fixed.